src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -159,7 +159,8 @@ public static partial class JsonSerializer
                     }
                 }
 
-                writer.WritePropertyName(key);
+                JsonEncodedText escapedKey = JsonEncodedText.Encode(key, options.Encoder);
```

Comment:
This is an incorrect change. It is undoing the optimizations from https://github.com/dotnet/corefx/pull/40991#discussion_r324953747 / https://github.com/dotnet/corefx/pull/41210

Please verify that nothing else from that PR is being changed, unintentionally.

Reply:
Thanks! Updated that and re-verified other commits.

Reply:
> Also, how did this diff fall through?

It didn't. That fix was done in an earlier change and was out-of-order and manually applied.